#163426613 Add ESlint configuration

mrkingz opened the pull request:
#### What does this PR do?
Add ESlint configuration to enable linting
#### Description of Task to be completed?
Enables linting to find problematic patterns or codes that don’t adhere to certain style guidelines
#### What are the relevant pivotal tracker stories?
[#163426613](https://www.pivotaltracker.com/n/projects/2240406/stories/163426613)

eltronix committed:
chore(setup) : Update travis CI to run test

- Encrypt travis slack notification string
- Add editorconfig file

[Finishes #163426611]

Commit:
chore(setup) link github with slack
[Delivers #163426609]

Commit:
chore(setup): integrate houndci with repo

- create a badge for houndci
- add the houndci badge to readme

[Delivers #163426614]

Commit:
chore(setup): fix dependency security vulnerability

- update the dependency to the lastest version

[Delivers #163567904]

Commit:
Merge pull request #9 from andela/ch-fix-dependency-vulnerability-163567904

#163567904 Fix Security Vulnerability in a Dependency

benfluleck commented on .eslintignore:
```diff
@@ -0,0 +1,2 @@
+# Dependency directory
+node_modules
```
No new line here

mrkingz replied:
Fixed. Thanks

benfluleck commented on .eslintrc.json:
```diff
@@ -0,0 +1,65 @@
+            }
+        }]       
+    }
+}
```
New line needed here

benfluleck commented:
There are too many commit messages in this PR. Please squash them @mrkingz

Commit:
Merge pull request #11 from andela/ch-add-travis-badge-163426611

#163426611 Enable travis CI and add travis badge

eokwukwe reviewed (changes requested):
Edit PR message. The relevant PT story part needs to only contain the story ID and it should be made to link to the PT story.

benfluleck commented on package.json:
```diff
@@ -3,8 +3,28 @@
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "node server/index.js",
+    "test": "exit 0",
+    "lint": "./node_modules/.bin/eslint 'server/**/*js'",
```
install eslint globally,  I expect `eslint '**/*js`
Why is this being done one by one. it should just be one command to fix
`    "lint:fix": "prettier-eslint '**/*js --write",`
Why is there so much consideration for json files, where are they coming from?

mrkingz replied:
Fixed

benfluleck commented:
@mrkingz  please squash and amend your  commits, there are too many

benfluleck commented on .gitignore:
```diff
@@ -28,13 +28,19 @@ build/Release
 # Optional REPL history
 .node_repl_history
 
+<<<<<<< HEAD
```
This happens when you have ignored your conflicts please fix

benfluleck commented on .gitignore:
```diff
@@ -28,13 +28,19 @@ build/Release
+=======
+# coveralls
+.coveralls.yml
+>>>>>>> chore (Setup): configure ESlint - Add ESlint configuration [Finishes #163426613]
```
This should not be showing up in your branch

mrkingz committed:
chore (Setup): ESlint configuration
- add Eslint configuration to enforce adherence to best practices and the approved coding convention.
[Finishes #163426613]
